apache-rat-core/src/main/java/org/apache/rat/analysis/TikaProcessor.java

```diff
@@ -133,6 +148,31 @@ public static String process(final Document document) throws RatDocumentAnalysis
         }
     }
 
+    /**
+     * Determine the character set for the input stream.  Input stream must implement mark.
+     * @param stream the stream to check.
+     * @return the detected characterset or null if not detectable.
+     * @throws IOException on IO error.
+     */
+    private static Charset detectCharset(final InputStream stream) throws IOException {
+        CharsetDetector encodingDetector = new CharsetDetector();
+        encodingDetector.setText(stream);
+        CharsetMatch charsetMatch = encodingDetector.detect();
+        if (charsetMatch != null) {
+            try {
+                return Charset.forName(charsetMatch.getName());
+            } catch (UnsupportedCharsetException e) {
+                // do nothing
```

Comment:
Should we at least log the exception here in order to debug "unexpected behaviour" when browsing contents?